### PR TITLE
[trello.com/c/bSmPAIM2] Feat: Show fixed date on scroll

### PR DESCRIPTION
**Pull request by IanaaDvlp**

### CODE REVIEW

File: Adamant/Modules/Chat/View/ChatViewController.swift [hunk 1]
```diff
@@ -713,6 +700,29 @@ private extension ChatViewController {
     func updateScrollDownButtonVisibility() {
         scrollDownButton.isHidden = isScrollPositionNearlyTheBottom
     }
+    
+    func updateDateHeaderIfNeeded() {
+        guard viewAppeared else { return }
+        
+        let targetY: CGFloat = targetYOffset + view.safeAreaInsets.top
+        guard let visibleIndexPaths = messagesCollectionView.indexPathsForVisibleItems.sorted(by: {
```
Comment (just-software-dev) [inline Adamant/Modules/Chat/View/ChatViewController.swift:708]: Why sorting?

File: Adamant/Modules/Chat/View/ChatViewController.swift [hunk 2]
```diff
@@ -713,6 +700,29 @@ private extension ChatViewController {
...
+            $0.row < $1.row
+        }) as [IndexPath]? else { return }
+        
+        for indexPath in visibleIndexPaths {
+            guard let cell = messagesCollectionView.cellForItem(at: indexPath)
```
Comment (just-software-dev) [inline Adamant/Modules/Chat/View/ChatViewController.swift:713]: Do we need to proceed with the highest visible cell only?

Comment (IanaaDvlp) [reply Adamant/Modules/Chat/View/ChatViewController.swift:713]: We go through the array and check if the cell matches our conditions

File: Adamant/Modules/Chat/View/ChatViewController.swift (hunk 1 above)
Comment (IanaaDvlp) [reply Adamant/Modules/Chat/View/ChatViewController.swift:708]: Removed

**Review by StanislavDevIOS — APPROVED** [review]

**Review by just-software-dev — APPROVED** [review]

**Pull request merged**